astropy/units/core.py

```diff
@@ -2094,14 +2094,24 @@ def __call__(
             if isinstance(s, bytes):
                 s = s.decode("ascii")
 
+            warning_actions = {"silent": "ignore", "warn": "default", "raise": "error"}
             try:
-                return f.parse(s)
+                warning_action = warning_actions[parse_strict]
+            except KeyError:
+                raise ValueError(
+                    "'parse_strict' must be 'warn', 'raise' or 'silent'"
+                ) from None
+            try:
+                with warnings.catch_warnings(
+                    action=warning_action, category=UnitParserWarning
+                ):
+                    return f.parse(s)
             except NotImplementedError:
                 raise
+            except UnitParserWarning as err:
+                raise ValueError(str(err))
```

Comment:
should this be `from err`? Or `from None`? 

Not perhaps really needed, but if we chain, we can also have some text here, perhaps noting that this error can be dealt with.

Reply:
We don't end up in this except block with critical errors that the parser couldn't handle, so adding a remark about `parse_strict` to the error message is a good idea.